flutter-studio/src/io/flutter/android/AndroidModuleLibraryManager.java

```diff
@@ -0,0 +1,198 @@
+    // TODO(messick): Collect URLs for all Android modules, including those within plugins.
+    VirtualFile dir = flutterProject.getBaseDir().findChild("android");
+    assert (dir != null);
+    EmbeddedAndroidProject androidProject = new EmbeddedAndroidProject(FileUtilRt.toSystemIndependentName(dir.getPath()), null);
+    androidProject.init();
+
+    GradleSyncListener listener = new GradleSyncListener() {
+      @SuppressWarnings("override")
+      public void syncTaskCreated(@NotNull Project project, @NotNull GradleSyncInvoker.Request request) {}
+
+      @Override
+      public void syncStarted(@NotNull Project project, boolean skipped, boolean sourceGenerationRequested) {}
+
+      @Override
+      public void setupStarted(@NotNull Project project) {}
+
+      @Override
+      public void syncSucceeded(@NotNull Project project) {
+        callback.apply(androidProject);
+        isUpdating.set(false);
+      }
+
+      @Override
+      public void syncFailed(@NotNull Project project, @NotNull String errorMessage) {
+        isUpdating.set(false);
+      }
+
+      @Override
+      public void syncSkipped(@NotNull Project project) {
+        isUpdating.set(false);
+      }
+    };
+
+    GradleSyncInvoker.Request request = GradleSyncInvoker.Request.projectLoaded();
+    request.runInBackground = true;
+    GradleSyncInvoker gradleSyncInvoker = ServiceManager.getService(GradleSyncInvoker.class);
+    gradleSyncInvoker.requestProjectSync(androidProject, request, listener);
+  }
+
+  @NotNull
+  public static AndroidModuleLibraryManager getInstance(@NotNull final Project project) {
+    return ServiceManager.getService(project, AndroidModuleLibraryManager.class);
+  }
+
+  public static void startWatching() {
+    // Start a process to monitor changes to Android dependencies and update the library content.
+    // This loop is for debugging, not production.
+    for (Project project : ProjectManager.getInstance().getOpenProjects()) {
+      if (project.isDefault()) {
+        continue;
+      }
+      if (FlutterSdkUtil.hasFlutterModules(project)) {
+        AndroidModuleLibraryManager manager = getInstance(project);
+        VirtualFileManager.getInstance().addVirtualFileListener(new VirtualFileContentsChangedAdapter() {
+          @Override
+          protected void onFileChange(@NotNull VirtualFile file) {
+            fileChanged(project, file);
+          }
+
+          @Override
+          protected void onBeforeFileChange(@NotNull VirtualFile file) {
+          }
+        }, project);
+
+        project.getMessageBus().connect().subscribe(ProjectTopics.PROJECT_ROOTS, new ModuleRootListener() {
+          @Override
+          public void rootsChanged(ModuleRootEvent event) {
+            manager.scheduleUpdate();
+          }
+        });
+        manager.scheduleUpdate();
+      }
+    }
+  }
+
+  private static void fileChanged(@NotNull final Project project, @NotNull final VirtualFile file) {
```

Comment:
Just for my own edification, it looks like we watch the `build.gradle` (in any subdirectory?), and schedule a gradle sync when it changes.

Reply:
Yes, there are two of them (at least) in an Android module. We might want to watch `settings.gradle` too.